src/rust/engine/src/nodes/execute_process.rs

```diff
@@ -124,6 +124,27 @@ impl ExecuteProcess {
                 .map_err(|e| format!("Failed to get `execution_slot_variable` for field: {e}"))?;
 
         let concurrency_available: usize = externs::getattr(value, "concurrency_available")?;
+        
+        let concurrency_value: Option<Bound<'_, PyAny>> = externs::getattr(value, "concurrency")?;
+        let concurrency: Option<ProcessConcurrency> = match concurrency_value {
+            None => Ok(None),
+            Some(conc) => {
+                let py_type = conc.get_type();
+                let py_name = py_type.name().unwrap();
+                let type_name = py_name.to_str().unwrap();
+                match type_name {
```

Comment:
Clunky, but I don't see a better way to model a rust enum in Python, so this is fine.

Reply:
The current pass at the python api feels clunky to use too though, I could use some perspective there in how we'd want plugin authors to define this (and eventually how it could trickle down into the fields of targets if necessary)